Source/Scene/Model.js

```diff
@@ -1267,6 +1292,79 @@ define([
+            return;
+        }
+
+        var extensions = model.gltf.extensions;
+        if (!defined(extensions) || !defined(extensions.mesh_compression_open3dgc)) {
+            return;
+        }
+
+        var decompressedBuffers = model.gltf.extensions.mesh_compression_open3dgc.decompressedBuffers;
+        while (loadResources.decompressedBuffersToCreate.length > 0) {
+            var name = loadResources.decompressedBuffersToCreate.dequeue();
+            var decompressedBuffer = decompressedBuffers[name];
+
+            var buffer = loadResources.buffers[decompressedBuffer.buffer];
+            var compressedBuffer = getSubarray(buffer, decompressedBuffer.byteOffset, decompressedBuffer.byteLength);
+
+            var decompressedArrayBuffer = new ArrayBuffer(decompressedBuffer.decompressedByteLength);
+            loadResources.decompressedBuffers[name] = new Uint8Array(decompressedArrayBuffer);
+
+            // Mostly copied from https://github.com/amd/rest3d/blob/master/server/o3dgc/js/index.php
+
+            var bstream = new o3dgc.BinaryStream(compressedBuffer);
+            var decoder = new o3dgc.SC3DMCDecoder();
+            var ifs = new o3dgc.IndexedFaceSet();
+
+            // Get metadata about the compression
+            decoder.DecodeHeader(ifs, bstream);
+
+            var len;
+            var decompressedBufferOffset = 0;
+
+            // Point the decoder at the TypedArrays it should write into
+            if (ifs.GetNCoordIndex() > 0) {
+                len = 3 * ifs.GetNCoordIndex();
+                ifs.SetCoordIndex(new Uint16Array(decompressedArrayBuffer, decompressedBufferOffset, len));
+                decompressedBufferOffset += len * Uint16Array.BYTES_PER_ELEMENT;
+            }
+            decompressedBufferOffset = alignOffset(decompressedBufferOffset, Float32Array.BYTES_PER_ELEMENT);
+            if (ifs.GetNCoord() > 0) {
+                len = 3 * ifs.GetNCoord();
+                ifs.SetCoord(new Float32Array(decompressedArrayBuffer, decompressedBufferOffset, len));
+                decompressedBufferOffset += len * Float32Array.BYTES_PER_ELEMENT;
+            }
+            if (ifs.GetNNormal() > 0) {
+                len = 3 * ifs.GetNNormal();
+                ifs.SetNormal(new Float32Array(decompressedArrayBuffer, decompressedBufferOffset, len));
+                decompressedBufferOffset += len * Float32Array.BYTES_PER_ELEMENT;
+            }
+            var numFloatAttributes = ifs.GetNumFloatAttributes();
+            for (var i = 0; i < numFloatAttributes; i++) {
+                if (ifs.GetNFloatAttribute(i) > 0) {
+                    len = ifs.GetFloatAttributeDim(i) * ifs.GetNFloatAttribute(i);
+                    ifs.SetFloatAttribute(i, new Float32Array(decompressedArrayBuffer, decompressedBufferOffset, len));
+                    decompressedBufferOffset += len * Float32Array.BYTES_PER_ELEMENT;
+                }
+            }
+            // TODO: handle int attributes?
```

Comment:
Perhaps add this to the TODO list in the pull request since we generally don't merge pull requests with TODO comments.